### PR TITLE
feat(atomic): add support for min and max for atomic-timeframe-facet

### PR DESCRIPTION
**Pull request by olamothe**
https://coveord.atlassian.net/browse/KIT-2108

### CODE REVIEW

**Review by ThibodeauJF — APPROVED** [review]

File: packages/atomic/src/components/search/facets/atomic-timeframe-facet/atomic-timeframe-facet.tsx [hunk 1]
```diff
@@ -142,6 +142,28 @@ export class AtomicTimeframeFacet
    */
   @MapProp() @Prop() public dependsOn: Record<string, string> = {};
 
+  /**
+   * The earliest date to accept when the `withDatepicker` option is enabled.
+   *
+   * This value must be a valid date string in the format `YYYY-MM-DD`.
+   *
+   * If the format is not respected, the datepicker will behave as if there was no `min` value set.
```
Comment (jpmarceau) [inline packages/atomic/src/components/search/facets/atomic-timeframe-facet/atomic-timeframe-facet.tsx:150]:
```suggestion
   * If this format is not respected, the date picker ignores this property, behaving as if no `min` value had been set.
```

File: packages/atomic/src/components/search/facets/atomic-timeframe-facet/atomic-timeframe-facet.tsx [hunk 2]
```diff
@@ -142,6 +142,28 @@ export class AtomicTimeframeFacet
...
+   * See also [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/date#min).
+   */
+  @Prop({reflect: true}) public min?: string;
+
+  /**
+   * The latest date to accept when the `withDatepicker` option is enabled.
+   *
+   * This value must be a valid date string in the format `YYYY-MM-DD`.
+   *
+   * If the format is not respected, the datepicker will behave as if there was no `max` value set.
```
Comment (jpmarceau) [inline packages/atomic/src/components/search/facets/atomic-timeframe-facet/atomic-timeframe-facet.tsx:161]:
```suggestion
   * If this format is not respected, the date picker ignores this property, behaving as if no `max` value had been set.
```

File: packages/atomic/src/components/search/facets/atomic-timeframe-facet/atomic-timeframe-facet.tsx [hunk 3]
```diff
@@ -142,6 +142,28 @@ export class AtomicTimeframeFacet
    */
   @MapProp() @Prop() public dependsOn: Record<string, string> = {};
 
+  /**
+   * The earliest date to accept when the `withDatepicker` option is enabled.
```
Comment (jpmarceau) [inline packages/atomic/src/components/search/facets/atomic-timeframe-facet/atomic-timeframe-facet.tsx:146]: My suggestions are all minor:
```suggestion
   * The earliest date to accept from user input when the `withDatepicker` option is enabled.
```

File: packages/atomic/src/components/search/facets/atomic-timeframe-facet/atomic-timeframe-facet.tsx [hunk 4]
```diff
@@ -142,6 +142,28 @@ export class AtomicTimeframeFacet
...
+   * This value must be a valid date string in the format `YYYY-MM-DD`.
+   *
+   * If the format is not respected, the datepicker will behave as if there was no `min` value set.
+   *
+   * See also [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/date#min).
+   */
+  @Prop({reflect: true}) public min?: string;
+
+  /**
+   * The latest date to accept when the `withDatepicker` option is enabled.
```
Comment (jpmarceau) [inline packages/atomic/src/components/search/facets/atomic-timeframe-facet/atomic-timeframe-facet.tsx:157]:
```suggestion
   * The latest date to accept from user input when the `withDatepicker` option is enabled.
```

**Review by jpmarceau — APPROVED** [review]
👌

**Review by btaillon — APPROVED** [review]

**Pull request merged**